GetStream/stream-chat-swift
# Add support for starting thread from oldest replies

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/351

### 🎯 Goal
Adding support for threads starting from the oldest replies instead of the newest. (Slack-like)

### 📝 Summary
Some extra stuff was implemented in this PR since it was related.
- Add support for threads starting from the oldest replies when `threadRepliesStartFromOldest` is enabled
- Rename `scrollToMostRecentMessage` -> `scrollToBottom`
- Rename `ScrollToLatestButton` -> `ScrollToBottomButton`
- Improved Channel List prefetching speed when loading more channels. (This one was not related but a quick change, one liner)

The renames were done because it was becoming very confusing. We had 3 names for the same thing. "ScrollToBottom", "ScrollToLatestMessage", and "ScrollToMostRecentMessage" 🤯 ... So it made sense to add consistency here, especially since we are introducing now a new "ScrollToTop".

### 🎨 Showcase

https://github.com/GetStream/stream-chat-swift/assets/12814114/527f0a5f-2002-42dc-b260-885b6d9b054f

### 🧪 Manual Testing Notes
- Open Demo App Configuration
- Enable both "shouldMessagesStartAtTheTop" and "threadRepliesStartFromOldest"
- Open a channel
- When opening a Thread, the replies should start from the oldest replies

### ☑️ Contributor Checklist
